src/lightning_lite/utilities/exceptions.py

```diff
@@ -12,6 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import Any, Type
 
-class MisconfigurationException(Exception):
-    """Exception used to inform users of misuse with Lightning."""
+from lightning_utilities.core.rank_zero import rank_zero_deprecation
+
+
+class _DeprecatedException(type):
+    def __instancecheck__(cls: Type, instance: object) -> bool:
+        # https://peps.python.org/pep-3119/
+        return any(cls.__subclasscheck__(c, stacklevel=7) for c in {type(instance), instance.__class__})
+
+    def __subclasscheck__(cls, subclass: Type, stacklevel: int = 5) -> bool:
+        mro = subclass.mro()
+        real = mro[1]
+        if cls is not subclass and cls is real:
+            raise TypeError(f"`{subclass.__name__}` should not be used for checks.")
+        if cls is MisconfigurationException:
+            rank_zero_deprecation(
+                f"`{MisconfigurationException.__name__}` is deprecated. Please check with `{real.__name__}` instead.",
+                stacklevel=stacklevel,
+            )
+        return cls in mro
+
+
+class MisconfigurationException(Exception, metaclass=_DeprecatedException):
```

Comment:
For files under `src/lightning_lite`, you can use the regular Python exceptions. No need to care about backwards compatibility.

cc @awaelchli 

The code you added here, should just be in `src/pytorch_lightning/utilities/exceptions.py`

Reply:
I'll revert this files' changes and move them under `pytorch_lightning`